### PR TITLE
Clarify that navigator.userAgentData is not mature

### LINKED ISSUE #11934: Issue with "Navigator.platform": premature recommendation to move to NavigatorUAData.platform
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform

#### What information was incorrect, unhelpful, or incomplete?
It's unhelpful to recommend NavigatorUAData.platform when it is not supported by several major browsers yet. At least without a caveat or suggested polyfill.

#### What did you expect to see?
The following would be better. Best would be some sort of polyfill or other recommendation for adopting this new approach.
>Note: The recommended alternative to this property is NavigatorUAData.platform, which is returned from Navigator.userAgentData. However, this property is not yet supported by some major browsers, and is still in the draft phase.

#### Did you test this? If so, how?
These two pages support my statement that the recommended alternative is still a draft and not yet supported by some major browsers.
I looked at https://wicg.github.io/ua-client-hints/#dom-navigatoruadata-platform and https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/platform

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/navigator/platform`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/navigator/platform/index.md
* Last commit: https://github.com/mdn/content/commit/45042595c38501cedb99e17e6b2d98d0451b7338
* Document last modified: 2021-10-21T23:35:07.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/11934

### CODE REVIEW

**Comment by wbamberg** [comment]
I thought it was weird to deprecate a feature when the replacement isn't available cross-browser or even standardised! Do you know where it is deprecated? https://html.spec.whatwg.org/multipage/system-state.html#dom-navigator-platform-dev doesn't say anything to that effect.

**Comment by sideshowbarker** [comment]
> I thought it was weird to deprecate a feature when the replacement isn't available cross-browser or even standardised! Do you know where it is deprecated? [html.spec.whatwg.org/multipage/system-state.html#dom-navigator-platform-dev](https://html.spec.whatwg.org/multipage/system-state.html#dom-navigator-platform-dev) doesn't say anything to that effect.

There’s been discussion about this elsewhere, but the gist of it is:

* this is one of key use cases for our proposed “Discouraged” language (vs “Deprecated”) — as discussed in https://github.com/mdn/content/discussions/5549
* developers should not `navigator.platform` because it’s not guaranteed to return what they’d expect.

Specifically, the spec allows browsers to simply return the empty string for`navigator.platform`. That’s not what developers would expect it to return, and it’s not useful in any way. So we want to warn developers away from trying to use it. That’s also why we have the existing MDN article saying:

> The specification allows browsers to always return the empty string, so don't rely on this property to get a reliable answer. 

Further, in general most everything in the `NavigatorID` mixin https://html.spec.whatwg.org/multipage/system-state.html#client-identification is effectively deprecated because:

* At the end of https://html.spec.whatwg.org/multipage/system-state.html#client-identification the spec has a big red note that ends with:
  > user agent implementers are strongly urged to include as little information in this API as possible.
* As with `navigator.platform`, most of the members are defined by the spec either as never returning any useful information at all, or else optionally returning something marginally useful, or nothing at useful at all.

In other words, the spec doesn’t deprecate the features but instead actually normatively defines requirements for them — however, the spec-defined requirements intentionally make the features effectively useless or unreliable to developers.

Hence we want to discourage developers from trying to use the features.

**Review by wbamberg — APPROVED** [review]
Thank you for the clarification @sideshowbarker !
